### PR TITLE
Add middleware

### PR DESCRIPTION
**Pull request by italolelis**
Adds a basic HTTP middleware and a context package

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -126,24 +103,24 @@ import (
         "net/http"
         "os"
 
-        "github.com/example/app/middleware"
-        "github.com/gin-gonic/gin"
+        "github.com/go-chi/chi"
```
Comment (vgarvardt) [inline README.md]: 👍

File: context/context.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+// New returns a context that has a stats Client
+func New(ctx context.Context, client client.Client) context.Context {
+	return context.WithValue(ctx, statsKey, client)
+}
+
+// WithContext returns a stats Client with as much context as possible
+func WithContext(ctx context.Context) client.Client {
+	ctxStats, ok := ctx.Value(statsKey).(client.Client)
+	if !ok {
+		return client.NewNoop(true)
```
Comment (vgarvardt) [inline context/context.go]: let's remove `unicode` parameter from `noop` client as it does not make any sense here

Comment (italolelis) [reply context/context.go]: so should I set it to false?

**Comment by vgarvardt** [comment]
👍

**Pull request merged**